app/src/main/java/ch/epfl/culturequest/SignUpActivity.java

```diff
@@ -20,35 +29,135 @@ public class SignUpActivity extends AppCompatActivity {
         Database.setPersistenceEnabled();
     }
 
+    EditText email, password;
+    TextView issues;
+    Button signUp, signIn;
+    String EMAIL_REGEX = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$";
+    String PW_REGEX = "^(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{8,}$"; // 8 chars long with at least 1 digit and at least 1 special char
+    TextWatcher passwordWatcher = new TextWatcher() {
+        @Override
+        public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+        }
+
+        @Override
+        public void onTextChanged(CharSequence s, int start, int before, int count) {
+            issues.setText("");
+            if (email.getText().toString().matches(EMAIL_REGEX)) {
+                setEnabled(s.toString().matches(PW_REGEX));
+            } else {
```

Comment:
Maybe add a comment to explain what happens if password and mail are correct, it is not clear when we read it like that